src/client/api/event.c

```diff
@@ -1222,12 +1222,25 @@ daos_event_abort(struct daos_event *ev)
 }
 
 int
-daos_event_priv_reset(void)
+daos_event_priv_reset(bool force)
 {
-	int rc;
+	int	retry = 0;
+	int	rc;
 
 	if (ev_thpriv_is_init) {
-		rc = daos_event_fini(&ev_thpriv);
+
+		do {
+			rc = daos_event_fini(&ev_thpriv);
+			if (rc != -DER_BUSY || !force)
+				break;
+			ev_thpriv.ev_abort = 1;
+			/*
+			 * Wait and retry at most 60 seconds if related task is in busy status.
+			 * Currently, it is used by test logic to cleanup task with loop retry.
+			 */
+			sleep(1);
```

Comment:
this is unacceptable imo. sleep should always be avoided in user space code. this can introduce many issues for apps.

Reply:
daos_event_priv_reset() is not application visible API, instead, it is only used by DAOS internal tests.

Reply:
i did not say that. my concern is that you are adding a new field in the public event structure (that users do not know about and will cause confusion if they try to use) and a sleep in the code just for testing purposes. this solution is not acceptable IMO.

Reply:
I re-checked the logs and related cmocka logic, and found that the test process has already long jumped out of related task context. Under such case, aborting task become meaningless. I will remake the patch to handle that.